### PR TITLE
Refactor/accept spdx expressions

### PR DESCRIPTION
**Pull request by neodmy**
## Description
Accept SDPX expressions for both `satisfies` arguments

## Motivation and Context
The original package doesn't allow `second` to be a SPDX expression, only a list of simple licenses. Users may want to compare plain SDPX expressions instead

## How Has This Been Tested?
Added new tests with Jest. Cases are included in `test/cases.js`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by nanotower — APPROVED** [review]
I'm fine with removing the second element as an array. This is not our use case and it would be as easy as passing a string with licenses separated by OR or AND. 
🆒

**Comment by neodmy** [comment]
> I'm fine with removing the second element as an array. This is not our use case and it would be as easy as passing a string with licenses separated by OR or AND. 🆒

@nanotower I have other functions in mind. Coming on the next PR

**Comment by neodmy** [comment]
> > I'm fine with removing the second element as an array. This is not our use case and it would be as easy as passing a string with licenses separated by OR or AND. 🆒
> 
> @nanotower I have other functions in mind. Coming on the next PR

I've decided to do it in this PR.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,14 +6,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: ['4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14']
+        node: ['14', '16', '18']
```
Comment (inigomarquinez) [inline .github/workflows/ci.yml]: Should we add node v20 as it's going to be the new LTS soon?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,14 +6,14 @@ jobs:
...
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node }}
           check-latest: true
       - run: npm install
-      - run: npm test
+      - run: npm run test:coverage
```
Comment (inigomarquinez) [inline .github/workflows/ci.yml]: Can we change the command to be `coverage` for consistency with other repos in tier 1?

File: package.json [hunk 3]
```diff
@@ -34,9 +35,8 @@
     "index.js"
   ],
   "scripts": {
-    "test": "npm run test:suite && npm run test:readme",
-    "test:suite": "node test.js",
-    "test:readme": "defence -i javascript README.md | replace-require-self | node",
+    "test": "jest",
+    "test:coverage": "jest --coverage",
```
Comment (inigomarquinez) [inline package.json]: can we change it to just `coverage`?

**Review by inigomarquinez — CHANGES_REQUESTED** [review]

**Review by inigomarquinez — APPROVED** [review]
En contra de mi voluntad, pero venga, vale...

**Comment by inigomarquinez** [comment]
@all-contributors please add @neodmy for code,doc,test

**Comment by inigomarquinez** [comment]
@all-contributors please add @nanotower for review

**Comment by inigomarquinez** [comment]
https://github.com/all-contributors please add @inigomarquinez  for review

**Comment by inigomarquinez** [comment]
@all-contributors please add @inigomarquinez for review

**Pull request merged**